python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -807,6 +807,31 @@ def test_datetime64_to_date32(self):
 
         assert arr2.equals(arr.cast('date32'))
 
+    def test_pandas_datetime_to_date64(self):
```

Comment:
Could you expand the test to also check the case where a non-None mask is passed?